### PR TITLE
Fix: ride hail transit re-enable

### LINKED ISSUE #3781: ridehai_transit mode is not working
After some debugging, it was found that the following is at fault: https://github.com/LBNL-UCB-STI/beam/commit/7de1559f89396c3a01f0853518ed810efa7c1a42

In order to fix this issue, change DRIVE_TRANSIT to RIDE_HAIL_TRANSIT here:
https://github.com/LBNL-UCB-STI/beam/blob/62aad326115fa6cf7380d8a16a0fee1a204db0ee/src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala#L655

Also need a test that reproduces the issue.

### PR DESCRIPTION
**Pull request by nikolayilyin**
- fixed RH transit mode
- added test to check all modes are working

- - -
This change is [Reviewable](https://reviewable.io/reviews/LBNL-UCB-STI/beam/3790)

### CODE REVIEW

**Comment by nikolayilyin** [comment]
fix for #3781

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

File: src/main/scala/beam/sim/population/PopulationAdjustment.scala [hunk 1]
```diff
@@ -56,6 +56,12 @@ trait PopulationAdjustment extends LazyLogging {
     populationWithAttributes
   }
 
+  // required for testing purposes
+  def logInfo(message: String): Unit = logger.info(message)
+
+  // required for testing purposes
+  def logError(message: String): Unit = logger.error(message)
```
Comment (dimaopen) [inline src/main/scala/beam/sim/population/PopulationAdjustment.scala:63]: Basically we can use it. Though it looks a little strange. What we can do is to create a custom Logback appender (https://www.baeldung.com/custom-logback-appender) and in logback-test.xml add it to the PopulationAdjustment only. Then we can get the appender programmatically and cleanup/check it before/after running a test.

**Review by dimaopen — APPROVED** [review]
If what I suggested seems too complicated feel free to skip it.

File: src/main/scala/beam/sim/population/PopulationAdjustment.scala (hunk 1 above)
Comment (nikolayilyin) [reply src/main/scala/beam/sim/population/PopulationAdjustment.scala:63]: Previously it did not work with custom appender and logger in the test (the approach was different though), and it looks like suggested change is too much work just to get one test working

**Pull request merged**